### ISSUE TITLE
[Question] I trained a model with Token_Classification_NER tutorial and saved it as .pt file how can I load up and use it in Pycharm? (or any python env to recommend)

### ISSUE THREAD

**Issue by 181473h** [NVIDIA/NeMo#1456]
What are the imports or anything I may need? I have pip install nemo_toolkit[nlp] installed already.

Do I need to (2. Define and initialize the neural network) as mentioned in https://pytorch.org/tutorials/recipes/recipes/saving_and_loading_models_for_inference.html

How may I do it?

Appreciate any help T.T Thanks in advance

**Comment by ericharper** [comment]
You may use the .pt file as any other Pytorch checkpoint or you can save your model as a .nemo file. The .nemo file will have everything needed (except datasets) to use your model. Please see the NeMo Primer tutorial for more information: https://colab.research.google.com/github/NVIDIA/NeMo/blob/v1.0.0b2/tutorials/00_NeMo_Primer.ipynb#scrollTo=g91FE8mlMcnh

**Comment by 181473h** [comment]
okay thank you!

**Comment by 181473h** [comment]
I got the file as .nemo now, how can I just pass in a text file with only some sentences inside to have it predict the labels for me?

As it does in this part of the tutorial,
`
# define the list of queries for inference
queries = [
    'we bought four shirts from the nvidia gear store in santa clara.',
    'Nvidia is a company.',
    'The Adventures of Tom Sawyer by Mark Twain is an 1876 novel about a young boy growing '
    + 'up along the Mississippi River.',
]
results = pretrained_ner_model.add_predictions(queries)

for query, result in zip(queries, results):
    print()
    print(f'Query : {query}')
    print(f'Result: {result.strip()}\n')`

**Comment by ekmb** [comment]
@181473h To load the pre-trained model:
model = TokenClassificationModel.restore_from("path_to_.nemo")
model.add_predictions(queries)

If you need to continue training, please follow the setup in the example: https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/token_classification/token_classification.py#L89

**Comment by 181473h** [comment]
What do I define the TokenClassificationModel with?
